### ISSUE TITLE
architecture: federated eval suites over canonical Zer0 fixtures

### ISSUE THREAD

**Issue by kvnloo** [kvnloo/z0evals#5]
## Goal

Make z0evals the **cross-repo evaluation composition, certification, and publication layer** for Zer0 without duplicating fixtures, schemas, training splits, runtime policy, or measurement semantics already owned by sibling repositories.

The core rule:

> Fixture semantics stay with the repo that owns the behavior. z0evals pins and composes those fixtures into immutable cross-system suites.

### Tokenomics

Owns measurement semantics and conformance fixtures:

- `spec/schemas/event.schema.json`
- `spec/schemas/experiment.schema.json`
- `spec/schemas/outcome.schema.json`
- `spec/semantic-conventions.yaml`
- `fixtures/conformance.json`
- `fixtures/z0int_receipt.json`
- `fixtures/kerdoios_observation.json`

z0evals should consume these for telemetry/outcome conformance. It must not redefine verified-success, experiment identity, treatment hashes, cost, token, or latency semantics.

### AODL

Owns contract legality / orchestration structure:

- `examples/valid/**`
- `examples/invalid/**`
- `examples/compile-stop/**`
- `tests/fixtures/**`
- AODL compiler/validator itself

Relevant current examples include fanout/fanin, bounded recursion, dependency cycles, hidden privilege, payment execution, verifier authority violations, and compile-stop cases.

z0evals should use these as deterministic structure/safety gates. It must not create a second AODL schema or local copy of these fixtures.

### z0intelligence

Owns cognition/model-policy benchmarks and model outputs:

- `benchmarks/data/**`
- `benchmarks/manifests/evaluation-matrix.jsonl`
- `benchmarks/manifests/metric-contract.json`
- `benchmarks/manifests/source-selection.jsonl`
- `results/raw/**` + `SHA256SUMS`
- benchmark runners/evaluators

z0evals can certify and publish cross-model results, but the model-policy benchmark corpus stays in z0intelligence.

### Evolution Lab

Owns experiment generation/search/training and locked learning splits.

Current P0 already has committed:

- `data/p0/train.npz`
- `data/p0/val.npz`
- `data/p0/confirm.npz`
- `data/p0/ood.npz`
- `data/p0/manifest.json`

and tests that prove the split is disk-locked rather than regenerated from RNG.

z0evals should normally use **confirm/OOD**, not train/val, for certification. Evaluation-only fixtures must not silently feed the training/distillation loop.

### Kerdoios

Owns allocation/placement behavior and its own component tests. Tokenomics already owns the cross-repo `kerdoios_observation.json` measurement fixture.

z0evals evaluates Kerdoios as a system-under-test; it should not own allocation logic.

## Proposed z0evals responsibility

z0evals owns only:

1. **suite manifests** that bind immutable source revisions + fixture paths;
2. **cross-repo adapters/runners** needed to compose those fixture providers;
3. **frozen result bundles** and artifact hashes;
4. **regression/non-inferiority gates** over those frozen suites;
5. **certification receipts** such as KEEP / DISCARD / PARTIAL;
6. **analysis/figures/public research posts** generated from the frozen evidence.

z0evals does NOT own:

- canonical component fixtures;
- training datasets;
- model routing/promotion policy;
- Tokenomics measurement semantics;
- AODL legality semantics;
- experiment search/mutation;
- provider/resource allocation;
- production runtime execution.

## Suite contract

A suite should pin every upstream source by immutable revision and content hash, for example:

```yaml
schema: z0eval.suite.v1
id: local-router-v1

sources:
  - repo: kvnloo/aodl
    revision: <40-char-sha>
    role: legality
    include:
      - examples/valid/fanout-fanin.json
      - examples/invalid/dependency-cycle.json
      - examples/invalid/hidden-privilege.json
      - examples/invalid/payment-execution.json

  - repo: kvnloo/tokenomics
    revision: <40-char-sha>
    role: measurement
    include:
      - fixtures/conformance.json
      - fixtures/z0int_receipt.json
      - spec/semantic-conventions.yaml

  - repo: kvnloo/z0intelligence
    revision: <40-char-sha>
    role: cognition
    include:
      - benchmarks/manifests/metric-contract.json
      - <owned routing fixture paths>

  - repo: kvnloo/evolution-lab
    revision: <40-char-sha>
    role: holdout
    include:
      - <confirm/ood manifest references>

data_policy:
  training_allowed: false
  split_roles: [confirm, ood]

gates:
  - contract_conformance
  - authority_safety
  - verified_quality_noninferiority
  - latency_tail
  - resource_regression
```

The runner materializes these refs into a local cache/worktree. Do not vendor copies into z0evals merely for convenience.

### L0 — Contract/conformance

Cheap deterministic gates:

- AODL valid examples compile
- AODL invalid / compile-stop examples are rejected
- Tokenomics events/receipts validate
- treatment hashes and experiment ids are stable
- artifact manifests/hash checks pass

### L1 — Component behavior

Evaluate the component on fixtures owned by that component:

- z0intelligence decision quality/calibration
- Kerdoios placement/allocation behavior
- OMP/Hermes adapter/tool lifecycle behavior
- JEV/OpenJev bounded decisions

### L2 — Cross-stack trajectory

Compose real system surfaces:

```text
AODL legal state
  -> z0intelligence route
  -> Kerdoios placement
  -> OMP/Hermes execution
  -> Tokenomics event/outcome
  -> independent verifier
```

Measure schema validity and trajectory correctness separately.

### L3 — Frozen confirmation / OOD

Use protected confirm/OOD groups with complete work-item lineage held together.

This is the promotion/regression firewall.

### L4 — Longitudinal canary

Compare current champion vs challenger under real shadow/canary traffic with stable Tokenomics semantics and tail-risk reporting.

Do not continuously rewrite the evaluator from production outcomes.

## Result contract

z0evals should emit a small certification receipt that points to, rather than duplicates, underlying evidence:

```yaml
schema: z0eval.result.v1
suite_id:
suite_revision:
system_revision:
baseline_revision:
verdict: KEEP | DISCARD | PARTIAL | NOT_COMPARABLE

gates:
  contract: pass
  safety: pass
  quality_noninferiority: pass
  latency_tail: pass

evidence:
  tokenomics_artifacts: []
  source_artifacts: []
  result_hashes: []

limitations: []
```

z0intelligence/Evolution Lab may consume this receipt for promotion logic. z0evals itself does **not** perform production promotion.

## Anti-leakage rule

Evaluation-only sources must be tagged and denied as training inputs by default.

For Q-Route/self-distillation:

```text
live/train receipts -> Evolution Lab / trainer
                     -> challenger

frozen confirm/OOD -> z0evals only
                   -> regression verdict
                   -> promotion evidence
```

The optimizer must not learn directly from the protected certification set after every failed run.

## Verified OSS Loop alignment

z0evals maps naturally to Verified OSS Loop Level 3: frozen evaluators produce KEEP/DISCARD evidence.

Evaluator/suite changes are equivalent to changing the oracle, so they should have stronger governance than CSS/blog changes:

- pin exact base/head revisions;
- evidence receipt on suite changes;
- independent review;
- sabotage/negative tests for evaluator changes;
- human merge authority;
- never silently update baseline thresholds when a challenger fails.

Recommend onboarding z0evals with a conservative rollout scheme and path-aware review for `suites/`, `schemas/`, and evaluator code.

## z0 registry integration

After the contracts above exist, add z0evals to `kvnloo/z0` as kind `evaluation`.

Conceptual edges:

```text
AODL ---------> z0evals  (contract fixtures)
Tokenomics ---> z0evals  (measurement/outcome semantics)
z0intelligence -> z0evals (candidate + cognition fixtures)
Evolution Lab -> z0evals  (experiment artifacts + protected holdouts)
Kerdoios -----> z0evals  (system under test)

z0evals ------> z0intelligence / Evolution Lab
                z0eval.result.v1 certification evidence
```

z0 remains the architecture registry; z0evals is not a second ecosystem registry.

## Acceptance

- [ ] document fixture-provider ownership matrix
- [ ] define `z0eval.suite.v1`
- [ ] define `z0eval.result.v1`
- [ ] implement revision-pinned source materialization/cache
- [ ] no copied Tokenomics/AODL/z0int fixtures in z0evals
- [ ] implement Tokenomics and AODL conformance adapters first
- [ ] enforce evaluation-only vs training-allowed data policy
- [ ] support grouped work-item/task-family holdouts
- [ ] add non-inferiority + tail-latency regression gates
- [ ] derive public posts/figures from frozen result artifacts
- [ ] onboard repo to Verified OSS Loop
- [ ] update z0 registry only after the cross-repo contract is real
